Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -391,8 +393,15 @@ private void SendDeadChat(EntityUid source, IPlayerSession player, string messag
     private void SendInVoiceRange(ChatChannel channel, string message, string messageWrap, EntityUid source, bool hideChat)
     {
         var sessions = new List<ICommonSession>();
+
         ClientDistanceToList(source, VoiceRange, sessions);
         _chatManager.ChatMessageToMany(channel, message, messageWrap, source, hideChat, sessions.Select(s => s.ConnectedClient).ToList());
+
+        foreach (var listenerEntity in ListenerEntityDistanceToList(source))
+        {
+            RaiseLocalEvent(listenerEntity, new ChatMessageHeardNearbyEvent(channel, message, messageWrap, source, hideChat));
```

Comment:
You shouldn't allocate a new event for every listener for this (also you need to create it out of the loop anyway to raise it by ref). Normally I'd be fine with the class event but the struct event is probably going out a lot so.